chain-impl-mockchain/src/chaineval.rs

```diff
@@ -38,7 +38,7 @@ impl AsRef<[u8]> for PraosNonce {
 
 /// Genesis Praos related data extract from the header
 #[derive(Debug, Clone)]
-pub(crate) struct HeaderGPContentEvalContext {
+pub struct HeaderGPContentEvalContext {
```

Comment:
Bikeshed: since it's going to be a public type, maybe name it more descriptively,
like `HeaderPraosContentEvelContext?`

Reply:
See the idea below about an enum.